src/webpack/loaders/devSitePropsTable.js

```diff
@@ -1,59 +1,97 @@
 const { getOptions } = require('loader-utils');
-const mdx = require('@mdx-js/mdx');
 const reactDocs = require('react-docgen');
 const findSource = require('../loaderUtils/findSource');
 
 /**
- * Create an inline mdx react component.
- * @param {*} description
- * @param {*} callback
+ * Parses prop types into a more compact representation.
+ * @param {object} type prop type information to parse.
+ * @param {string} type.value the value for the type
+ * @param {string} type.name the name of the type
+ * @param {boolean} type.required whether or not the type is required
+ * @param {description} type.description the description of the type
+ * @param {bool} indent should indent the return value.
  */
-const propMdx = async function propMdx(description, options, callback) {
-  const mdxOptions = {
-    ...options,
-    filepath: this.resourcePath,
-    skipExport: true,
-  };
+const parsePropType = ({ type, indent = false }) => {
+  const {
+    value,
+    name,
+    required,
+    description,
+  } = type;
+  let returnValue = [];
+
+  if (value) {
+    if (name === 'instanceOf') {
+      returnValue = [
+        `InstanceOf: ${value},`,
+      ];
+    } else if (name === 'enum') {
+      returnValue = [
+        'enum: [',
+        ...value.map(val => `  ${val.value}`),
+        '],',
+      ];
+    } else if (name === 'union') {
+      returnValue = [
+        'unionOf: [',
+        ...value.map((val) => parsePropType({ type: val, indent: true })),
+        '],',
+      ];
+    } else if (name === 'arrayOf') {
+      returnValue = [
+        'arrayOf: [{',
+        ...parsePropType({ type: value, indent: true }),
+        '}],',
+      ];
+    } else if (name === 'objectOf') {
+      returnValue = [
+        'objectOf: {',
+        ...parsePropType({ type: value, indent: true }),
+        '},',
+      ];
+    } else if (name === 'shape') {
+      returnValue = [
+        'shape: {',
+        ...Object.entries(value).reduce((acc, [key, val]) => acc.concat([`${key}: {`, ...parsePropType({ type: val, indent: true }), '},']), []).map((val) => `  ${val}`),
+        '},',
+      ];
+    } else if (name === 'exact') {
+      returnValue = [
+        'exactShape: {',
+        ...Object.entries(value).reduce((acc, [key, val]) => acc.concat([`${key}: {`, ...parsePropType({ type: val, indent: true }), '},']), []).map((val) => `  ${val}`),
```

Comment:
![https://media1.giphy.com/media/3o7btPCcdNniyf0ArS/200.gif](https://media1.giphy.com/media/3o7btPCcdNniyf0ArS/200.gif)

I trust you

Reply:
![](https://media.giphy.com/media/1KoN1DMBnCMWk/giphy.gif)